sh/tests/in-back-ground_test.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+# Unofficial Bash Strict Mode
+set -euo pipefail
+IFS='
+'
+
+FILENAME="student/in-back-ground.sh"
+script_dirS=$(cd -P "$(dirname "$BASH_SOURCE")" &>/dev/null && pwd)
+
+challenge() {
+    submitted=$(cd "$1" && bash "$script_dirS"/$FILENAME)
+    expected=$(cd "$1" && bash "$script_dirS"/solutions/in-back-ground.sh)
+    diff <(echo "$submitted") <(echo "$expected")
+    diff sol-output.log output.log
+}
+
+# True if FILE exists and is a regular file
+if [ -f ${FILENAME} ]; then
+    # FILE exists and it's not empty
+    if [ -s ${FILENAME} ]; then
+        challenge .
```

Comment:
should we add a test for a case where `grep` fails?

Reply:
i've added a test case for a fail.

Reply:
The fail case implemented is at the first `cat`, what about when `grep` fails? Should it behave in the same way?

Reply:
In this exercise we are testing for the usage of the nohup command, that is the purpose, but i've added something that will test when the grep doesn't catches anything.

About the message:
```
nprimo@nprimo-01pc:~/01/public/sh/tests$ bash in-back-ground_test.sh 
nohup: redirecting stderr to stdout
nohup: redirecting stderr to stdout
nohup: redirecting stderr to stdout
nohup: redirecting stderr to stdout
```
When the nohup command is used, it sends a message to the console indicating that it has redirected stderr to stdout, it is a normal behavior of the command and it can be ignored. we can prevent the message from being printed by redirecting stderr to "/dev/null" but the goal is to teach the student how to use the nohup command, and as this is the normal behavior i didn't ask to deal with the message.